### PR TITLE
Add settings for logging

### LINKED ISSUE #18: Accessible way to find log directory
Finding where Xi-CAM logs files to requires some digging into source code, which is not user-friendly. You can ask Xi-CAM where it is by running:

```python
from xicam.core.paths import user_cache_dir
print(f"{user_cache_dir}/logs")
```

We should at the very least expose this path through the Xi-CAM interface (e.g. its settings).

It would also be nice to be able to configure this log directory (the naming, overwrite, append, newfile, location, etc.) through the settings.

### LINKED ISSUE #19: Configurable logging base level
Add a way to configure the base log level in Xi-CAM's settings (i.e. expose `logging.setLevel` through the settings).

### PR DESCRIPTION
**Pull request by birdzix**
This PR adds a way to configure the logging level of xicam logger's StreamHandler. This gives the user an opportunity to select among the log levels provided through python's logging framework (default is `DEBUG`).

In addition, this PR exposes the directory path for the xicam logger's FileHandler in an system-agnostic manner. Note that this is *not* editable by the user; it is intended to be informational only.

### TODO

- [x] Update logo? -- *add an issue for this*  (#68).

### Questions / concerns

<details>
<summary>(addressed -- logging levels (stream and file) are configurable. Tool tips are enabled for the list parameter. file log level has a "recommended" mark on DEBUG)</summary>
* Do we want to expose the bare python logging levels as options (e.g. DEBUG, INFO, ...)?
  * If yes, we should consider adding some way to document what these may mean
  * If no, what should we use instead?
> Possible solutions: add some documentation into the Logging settings that explain what the parameters shown are.
</details>

<details>
<summary>(removed -- we do not want to do this.)</summary>
* The commit ad061c2 loads in the SettingsPlugins entry points *first*, *then* restores the `ConfigDialog` before the `collectPlugins()` call is made. **Is there any issue to doing this?** I did this to make sure *all* settings are loaded and applied before anything else is loaded. The reason I feel I needed to do this is that the SettingsPlugins are plugins; they are not part of the Xi-CAM library/application itself, and I want them to be loaded before any other plugins. Questions I have about this are:
  * Do the SettingsPlugins need to be plugins (I think maybe, since cammart and venvs are configurable via CLI arguments)?
  * Does the LoggingSettingsPlugin need to be a plugin?
</details>

<details>
<summary> (removed -- we do want the restore here, e.g. device settings in SAXS use this)</summary>
* The commit 9a3526b removes the `self.restore()` call inside of `ConfigDialog.show()`. The rationale for this is that when accessing the settings through `File -> Settings`, the settings *should* have already been loaded and applied to the session.
</details>

<details>
<summary> (removed -- just handling logging settings for now is fine) </summary>
* Should this be a "GenericSettings" Plugin? For example, you can modify the log level, inspect the log directory location, inspect where intake is looking for yml files, etc.?
</details>

<details>
<summary> (removed -- yes, you should be able to define settings plugins outside of Xi-CAM's primary packages)</summary>
* Can (should) SettingsPlugins be defined outside of Xi-CAM's primary packages, e.g. SAXS's `DeviceProfiles`?  Note this is a yaspy plugin does not have an entry point. What is the intent of `DeviceProfiles`? Should be configurable in both the Calibration stage and in Xi-CAM's settings?
</details>

### Related PRs
* https://github.com/synchrotrons/Xi-cam.core/pull/20
* https://github.com/synchrotrons/Xi-cam.plugins/pull/30/

### Issues addressed
Fixes https://github.com/synchrotrons/Xi-cam.core/issues/19
Fixes https://github.com/synchrotrons/Xi-cam.core/issues/18

### CODE REVIEW

File: xicam/gui/windows/mainwindow.py [hunk 1]
```diff
@@ -50,13 +50,16 @@ def __init__(self):
         # Setup appearance
         self.setWindowTitle("Xi-cam")
 
-        # Load plugins
-        pluginmanager.collectPlugins()
+        # First load Settings Plugins!
+        pluginmanager.load_entry_point_plugins(categories=['SettingsPlugin'])
```
Comment (birdzix) [inline xicam/gui/windows/mainwindow.py]: do not do this, use an alternate approach here

File: xicam/gui/windows/mainwindow.py [hunk 2]
```diff
@@ -50,13 +50,16 @@ def __init__(self):
...
 
         # Restore Settings
         self._configdialog = ConfigDialog()
         self._configdialog.restore()
```
Comment (birdzix) [inline xicam/gui/windows/mainwindow.py]: try removing this restore()

File: xicam/gui/windows/settings.py [hunk 3]
```diff
@@ -84,7 +84,9 @@ def show(self):
         if self.lastwidget:
             self.pagesWidget.addWidget(self.lastwidget)
             self.pagesWidget.setCurrentWidget(self.lastwidget)
-        self.restore()
+        # Do we need this? The settings should already be loaded by when requesting a show
+        # (see XicamMainWindow.__init__, where the settings menu item is created and connected)
+        # self.restore()
```
Comment (birdzix) [inline xicam/gui/windows/settings.py]: we need this restore() (DeviceProfiles needs this...)

**Comment by birdzix** [comment]
Note that these new changes *do* restore the settings correctly (e.g. if you change the appearance to a dark theme); however, the main window will load plugins, and *then* restore the settings (so the gui will look like its default theme, then switch to dark after all plugins have finished)

**Comment by birdzix** [comment]
The file log level is now configurable.
For the file log levels, marked "DEBUG" as recommended in the drop down list. We should eventually add better descriptions into the logging settings plugin, but I think this is fine for now.

File: xicam/gui/windows/mainwindow.py [hunk 4]
```diff
@@ -50,12 +50,13 @@ def __init__(self):
         # Setup appearance
         self.setWindowTitle("Xi-cam")
 
+        self._configdialog = ConfigDialog()
+
+        # Restore Settings
+        pluginmanager.attach(self._configdialog.restore)
```
Comment (ronpandolfi) [inline xicam/gui/windows/mainwindow.py]: Move into configdialog?

File: xicam/gui/settings/logging.py [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+                ),
+            ],
+        )
+        msg.file_handler.setLevel(self["File Log Level"])
+        msg.stream_handler.setLevel(self["Terminal Log Level"])
+
+    def apply(self):
+        msg.file_handler.setLevel(self["File Log Level"])
+        msg.stream_handler.setLevel(self["Terminal Log Level"])
+        QSettings().setValue(msg.FILE_LOG_LEVEL_SETTINGS_NAME, self["File Log Level"])
```
Comment (ronpandolfi) [inline xicam/gui/settings/logging.py]: ParameterSettingsPlugin does its own serialize/deserialize. Is this redundant?

**Comment by birdzix** [comment]
Commit 56532cb addresses the following:

* log levels no longer re-serialized explicitly in `LoggingSettingsPlugin.apply`
* `ConfigDialog.__init__` is now responsible for attaching to `ConfigDialog.restore` when the plugin manager has finished loading plugins (instead of in `mainwindow`)
* fixes the default values for the logging parameters

**Review by ronpandolfi — APPROVED** [review]

**Pull request merged**